crates/starknet-devnet/tests/test_blocks_on_demand.rs

```diff
@@ -81,20 +105,36 @@ mod blocks_on_demand_tests {
             .await;
 
         assert_latest_block_with_transactions(&devnet, 1, tx_hashes).await;
-        assert_pending_block_with_transactions(&devnet, 2, 0).await;
+        assert_pending_block_with_transactions(&devnet).await;
+
+        assert_pending_state_update(&devnet).await;
+        assert_latest_state_update(&devnet, BlockId::Tag(BlockTag::Latest)).await;
     }
 
     #[tokio::test]
-    async fn pending_block_in_normal_mode() {
+    async fn pending_block_and_state_in_normal_mode() {
         let devnet = BackgroundDevnet::spawn().await.unwrap();
 
         devnet.create_block().await.unwrap();
 
-        let pending_block = devnet.get_pending_block_with_tx_hashes().await.unwrap();
+        let pending_block = devnet
+            .json_rpc_client
+            .get_block_with_tx_hashes(BlockId::Tag(BlockTag::Pending))
+            .await
+            .unwrap();
```

Comment:
Why not use `devnet.get_pending_block_with_tx_hashes()` ?

Reply:
oh, forgot that it needs to be like this because of a later one line assert:
`assert_eq!(block, latest_block);`